python/lsst/sphgeom/__init__.py

```diff
@@ -37,3 +38,130 @@
+    if name.startswith("__") and (
+        value is getattr(object, name, None) or name in _INTRINSIC_SPECIAL_ATTRIBUTES
+    ):
+        return False
+    return True
+
+
+def _continueClass(cls):
+    import sys
+
+    orig = getattr(sys.modules[cls.__module__], cls.__name__)
+    for name in dir(cls):
+        # Common descriptors like classmethod and staticmethod can only be
+        # accessed without invoking their magic if we use __dict__; if we use
+        # getattr on those we'll get e.g. a bound method instance on the dummy
+        # class rather than a classmethod instance we can put on the target
+        # class.
+        attr = cls.__dict__.get(name, None) or getattr(cls, name)
+        if _isAttributeSafeToTransfer(name, attr):
+            setattr(orig, name, attr)
+    return orig
+
+
+def _inf_to_lat(lat: float) -> float:
+    """Map +Inf to +90 and -Inf to -90 degrees."""
+    import math
+
+    if not math.isinf(lat):
+        return lat
+    if lat > 0.0:
+        return 90.0
+    return -90.0
+
+
+@_continueClass
+class Region:
+    """A minimal interface for 2-dimensional regions on the unit sphere."""
+
+    @classmethod
+    def from_ivoa_pos(cls, pos: str) -> Region:
+        """Create a Region from an IVOA POS string.
+
+        Parameters
+        ----------
+        pos : `str`
+            A string using the IVOA SIAv2 POS syntax.
+
+        Returns
+        -------
+        region : `Region`
+            A region equivalent to the POS string.
+
+        Notes
+        -----
+        See
+        https://ivoa.net/documents/SIA/20151223/REC-SIA-2.0-20151223.html#toc12
+        for a description of the POS parameter but in summary the options are:
+
+        * ``CIRCLE <longitude> <latitude> <radius>``
+        * ``RANGE <longitude1> <longitude2> <latitude1> <latitude2>``
+        * ``POLYGON <longitude1> <latitude1> ... (at least 3 pairs)``
+
+        Units are degrees in all coordinates.
+        """
+        shape, *coordinates = pos.split()
+        coordinates = tuple(float(c) for c in coordinates)
+        n_floats = len(coordinates)
+        if shape == "CIRCLE":
+            if n_floats != 3:
+                raise ValueError(f"CIRCLE requires 3 numbers but got {n_floats} in '{pos}'.")
+            center = LonLat.fromDegrees(coordinates[0], coordinates[1])
+            radius = Angle.fromDegrees(coordinates[2])
+            return Circle(UnitVector3d(center), radius)
+        elif shape == "RANGE":
+            import math
+
+            if n_floats != 4:
+                raise ValueError(f"RANGE requires 4 numbers but got {n_floats} in '{pos}'.")
+            # POS allows +Inf and -Inf in ranges. These are not allowed by
+            # Box and so must be converted.
+            # If either of the longitude values are infinite, all longitudes
+            # should be included.
+            if math.isinf(coordinates[0]) or math.isinf(coordinates[1]):
```

Comment:
So even if both longitudes are the same sign `Inf` it still translates to 0-360? The standard does not say which combinations of `Inf` are allowed here, maybe something like this does not make sense. `RANGE -Inf 90 ...` seems to be converted to `RANGE 0 360 ...`? Standard also mentions that `NaN` is valid, but does not say what it means, and we do not handle it here?

Reply:
Indeed. I was staring at the examples and trying to work out what the meant. Maybe `90 +Inf` is meant to mean `90 360`. With wraparound I was taking any Inf in longitude to mean the full range. Maybe @gpdf knows what they really meant. I also don't know what `NaN` is meant to do.

Reply:
James Dempsey has clarified that -Inf means 0 and +Inf means 360 so I've updated the code. He also stated that NaN is no longer part of the standard.